src/execution/compiler/operator/hash_join_translator.cpp

```diff
@@ -61,6 +63,26 @@ void HashJoinTranslator::DefineHelperStructs(util::RegionVector<ast::StructDecl
   decls->push_back(struct_decl);
 }
 
+void HashJoinTranslator::DefineHelperFunctions(util::RegionVector<ast::FunctionDecl *> *decls) {
```

Comment:
So I'm not sure how to get the table vector which corresponds to the right table row from within `HashJoinTranslator`, if I can get access to that then I can refactor `outerJoinConsumer` to take both the buildRow and the rightRow as arguements. This isn't an issue right now, because since I'm only calling `outerJoinConsumer` for the left outer join specfic portion - the right values are always NULL. 

(Also I would prefer refactoring this as well, not only for the code duplication, but would be easier if someone wanted to implement right outer join or outer join later)

Reply:
In this case, materialize the right row and call `outerJoinConsumer`. Reduce duplication in generated code.